File: solutions/go/resistor-color/1/resistor_color.go
package resistorcolor

import "maps"

var colors = map[string]int{
	"black":  0,
	"brown":  1,
	"red":    2,
	"orange": 3,
	"yellow": 4,
	"green":  5,
	"blue":   6,
	"violet": 7,
	"grey":   8,
	"white":  9,
}

// Colors returns the list of all colors.
func Colors() (list []string) {
	for i := range maps.Keys(colors) {
		list = append(list, i)
	}
	return
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	return colors[color]
}

File: solutions/python/resistor-color/1/resistor_color.py
from enum import IntEnum, auto


class Color(IntEnum):
    black = 0
    brown = auto()
    red = auto()
    orange = auto()
    yellow = auto()
    green = auto()
    blue = auto()
    violet = auto()
    grey = auto()
    white = auto()


def color_code(color: str) -> int:
    return Color[color].value


def colors() -> list[str]:
    return [c.name for c in Color]
